### DISCUSSION TITLE
The actual memory footprint of static_map is larger than my estimation

### DISCUSSION THREAD

**Discussion by Gibyeng** [NVIDIA/cuCollections#650] [Q&A] [answered]
Hi, I find the memory footprint of cuco::static_map is significantly larger than its theoretical size (`capacity* (sizeof(KEY) + sizeof(VALUE))`). For instance, I set up a static_map and its theoretical memory footprint is 4.65G. However, it actually consumes 6.20G GPU memory. Why is that?

**Comment by Gibyeng** [comment]
After a few tests, I find the real reason is memory alignment issue.

**Reply by PointKernel** [reply, marked as answer]
The alignment of your key and payload types can be one factor, while another is that cuco hash tables ensure the requested capacity serves as a lower bound. They select the smallest prime number greater than the requested capacity as the actual capacity. see https://github.com/NVIDIA/cuCollections/blob/93d61722e7c04f6504a7707830864affc03e1d0e/include/cuco/extent.cuh#L95-L99
